0.X/Dockerfile

```diff
@@ -50,10 +51,13 @@ RUN set -eux; \
 # storage backend, if desired; the server will be started with /vault/config as
 # the configuration directory so you can add additional config files in that
 # location.
+ENV HOME /home/vault
 RUN mkdir -p /vault/logs && \
     mkdir -p /vault/file && \
     mkdir -p /vault/config && \
-    chown -R vault:vault /vault
+    mkdir -p $HOME && \
+    chgrp -R 0 $HOME && chmod -R g+rwX $HOME && \
+    chgrp -R 0 /vault && chmod -R g+rwX /vault
```

Comment:
Looks like there's some chown'ing to vault:vault in the entrypoint script, do those need to be changed to vault:root? https://github.com/hashicorp/docker-vault/blob/5ae9f14c1f0b3770d400fc6a03235f888c2fdfcd/0.X/docker-entrypoint.sh#L72-L85

Reply:
This is guarded against by the environment variable `SKIP_CHOWN`: https://github.com/hashicorp/docker-vault/blob/5ae9f14c1f0b3770d400fc6a03235f888c2fdfcd/0.X/docker-entrypoint.sh#L71

There's a chance users are still running this as root in Docker, but we might consider removing it.

Reply:
Yeah, that skips the chown's if SKIP_CHOWN is set, but it looks to me that they run by default? If I build this image as `openshift_ent`:

```console
➜ docker run -it openshift_ent 
chown: /vault/config: Operation not permitted
chown: /vault/config: Operation not permitted
Could not chown /vault/config (may not have appropriate permissions)
chown: /vault/logs: Operation not permitted
chown: /vault/logs: Operation not permitted

➜ echo $?
1

```